cita-executor/core/src/executive.rs

```diff
@@ -326,11 +326,16 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 
         let schedule = Schedule::new_v1();
 
+        let gas_used_by_data_length = match t.version {
+            0...2 => 0,
+            _ => t.data.len() * 4,
+        };
+
         let base_gas_required = match t.action {
             Action::Create => schedule.tx_create_gas,
             Action::GoCreate => schedule.tx_create_gas,
             _ => schedule.tx_gas,
-        };
+        } + gas_used_by_data_length;
```

Comment:
This should be `+ gas_used_by_data_length * schedule.create_data_gas;` ?

Reply:
Why `* schedule.create_data_gas`?

Reply:
`store_gas_used = U256::from(t.data.len() * schedule.create_data_gas);`

Reply:
`store_gas_used = U256::from(t.data.len() * schedule.create_data_gas);`

Reply:
It's base gas instead of the store gas.  stored gas used will be calculated in vm.

> // gas_prepare = 21000 + (68 or 4 per byte) + (32000 if tx.to == 0)

Reply:
I think t.data.len() * schedule.create_data_gas is ok. 
Now, schedule.create_data_gas is 200, default BQL is 0x4000000, 0x4000000 / 200 = 5368709.12.

Reply:
The difference:
* one is before vm, and one is in vm
* one is tx data, and one will be stored in storage

What i want to say is that the base gas is not related to the instruction of vm.

Of course, We can define our rule about this or follow the Ethereum.

Reply:
But we also use `200` for each byte of our "存证" data. Should we follow this rule?

Reply:
In my humble opinion, stored data is actually tx data, it should cost base gas(ignore the already implement): 
```
21000 + (68 per byte)
```

Reply:
Okay! I agree with you!

Reply:
I have changed it to 21000 + (68 per byte)